Platforms/Export-Import-Platform.ps1

```diff
@@ -110,136 +129,214 @@ Function Join-ExceptionMessage
+Function import-platform {
+
+	param(
+		[string]$PlatformZipPath
+	)
+	If (Test-Path $PlatformZipPath) {
+		$zipContent = [System.IO.File]::ReadAllBytes($(Resolve-Path $PlatformZipPath))
+		$importBody = @{ ImportFile = $zipContent; } | ConvertTo-Json -Depth 3 -Compress
+		try {
+			$ImportPlatformResponse = Invoke-RestMethod -Method POST -Uri $URL_ImportPlatforms -Headers $logonHeader -ContentType "application/json" -TimeoutSec 2700 -Body $importBody
+			Write-Debug "Platform ID imported: $($ImportPlatformResponse.PlatformID)"
+			Write-Host "Retrieving Platform details"
+			# Get the Platform Name
+			$platformDetails = Invoke-RestMethod -Method Get -Uri $($URL_PlatformDetails -f $ImportPlatformResponse.PlatformID) -Headers $logonHeader -ContentType "application/json" -TimeoutSec 2700
+			If ($platformDetails) {
+				Write-Debug $platformDetails
+				Write-Host "$($platformDetails.Details.PolicyName) (ID: $($platformDetails.PlatformID)) was successfully imported and $(if($platformDetails.Active) { "Activated" } else { "Inactive" })"
+				Write-Host "Platform details:" 
+				$platformDetails.Details | Select-Object PolicyID, AllowedSafes, AllowManualChange, PerformPeriodicChange, @{Name = 'AllowManualVerification'; Expression = { $_.VFAllowManualVerification } }, @{Name = 'PerformPeriodicVerification'; Expression = { $_.VFPerformPeriodicVerification } }, @{Name = 'AllowManualReconciliation'; Expression = { $_.RCAllowManualReconciliation } }, @{Name = 'PerformAutoReconcileWhenUnsynced'; Expression = { $_.RCAutomaticReconcileWhenUnsynched } }, PasswordLength, MinUpperCase, MinLowerCase, MinDigit, MinSpecial 
+			}		
+		} catch {
+			#Write-Error $_.Exception
+			#Write-Error $_.Exception.Response
+			#Write-Error $_.Exception.Response.StatusDescription
+			
+			($_.ErrorDetails | ConvertFrom-Json | Select-Object -Property ErrorMessage)
+			"Error while attempting to export $PlatformZipPath"
+			""
+		}
+	}
+} #end function Import
+
+Function export-platform {
+
+	param(
+		[string]$PlatformID
+	)
+
+	try {
+		$exportURL = $URL_ExportPlatforms -f $PlatformID
+		Invoke-RestMethod -Method POST -Uri $exportURL -Headers $logonHeader -ContentType "application/zip" -TimeoutSec 2700 -OutFile "$PlatformZipPath\$PlatformID.zip" -ErrorAction SilentlyContinue
 	} catch {
-		Write-Error "Could not change SSL validation"
-		Write-Error (Join-ExceptionMessage $_.Exception) -ErrorAction "SilentlyContinue"
-		return
+		#Write-Error $_.Exception.Response
+		#Write-Error $_.Exception.Response.StatusDescription
+		
+		($_.ErrorDetails | ConvertFrom-Json | Select-Object -Property ErrorMessage)
+		"Error while attempting to export $PlatformID"
+		""
 	}
-}
-Else
-{
-	try{
-		Write-Debug "Setting script to use TLS 1.2"
-		[System.Net.ServicePointManager]::SecurityProtocol = [System.Net.SecurityProtocolType]::Tls12
+} #end function Import
+
+Function Get-PlatformsList {
+	param(
+		[switch]$GetAll
```

Comment:
As I see it, "GetAll" is always true and the switch here is if we want to have only the Active one or not
So maybe would make more sense to change the switch parameter name to "Only Active" or something like that, and change the If below accordingly